### PR TITLE
sip-30 Deprecate erc223 

### PR DESCRIPTION
**Pull request by hav-noms**
https://sips.synthetix.io/sips/sip-30

### CODE REVIEW

File: contracts/Depot.sol [hunk 1]
```diff
@@ -45,7 +45,7 @@ contract Depot is SelfDestructible, Pausable {
     using SafeDecimalMath for uint;
 
     /* ========== STATE VARIABLES ========== */
-    ISynthetix public synthetix;
+    address public snxProxy;
```
Comment (hav-noms) [inline contracts/Depot.sol]: use snxProxy here instead to remove call from update script

File: contracts/Depot.sol [hunk 2]
```diff
@@ -420,7 +420,7 @@ contract Depot is SelfDestructible, Pausable {
         fundsWallet.transfer(msg.value);
 
         // And send them the SNX.
-        synthetix.transfer(msg.sender, synthetixToSend);
+        IERC20(snxProxy).transfer(msg.sender, synthetixToSend);
```
Comment (hav-noms) [inline contracts/Depot.sol]: Don't need ISynthetix.sol its only doing a snx.transfer

File: contracts/Depot.sol [hunk 3]
```diff
@@ -549,43 +549,27 @@ contract Depot is SelfDestructible, Pausable {
...
-     *         if they'd like. You can equally just transfer synths to this contract and it will work
-     *         exactly the same way but with one less call (and therefore cheaper transaction fees)
+     * @notice depositSynths: Allows users to deposit synths via the approve / transferFrom workflow     
      * @param amount The amount of sUSD you wish to deposit (must have been approved first)
      */
     function depositSynths(uint amount)
         external
     {
-        // Grab the amount of synths
+        // Grab the amount of synths. Will fail if not approved first
```
Comment (hav-noms) [inline contracts/Depot.sol]: mintr must now call depositSynths()

File: contracts/Synthetix.sol [hunk 4]
```diff
@@ -276,68 +275,38 @@ contract Synthetix is ExternStateToken {
...
-     */
-    function transfer(address to, uint value, bytes data)
         public
         optionalProxy
         returns (bool)
     {
-        // Ensure they're not trying to exceed their locked amount
-        require(value <= transferableSynthetix(messageSender), "Insufficient balance");
+        // Ensure they're not trying to exceed their staked SNX amount
+        require(value <= transferableSynthetix(messageSender), "Cannot transfer staked or escrowed SNX");
```
Comment (hav-noms) [inline contracts/Synthetix.sol]: Message is longer that 32 bytes but its worth while message. hopefull will reduce support. I've save bytecode space in other messages and requires

File: contracts/Depot.sol (hunk 3 above)
Comment (jacko125) [reply contracts/Depot.sol]: Looks good, dapp would need to let users to `approve()` first.

File: contracts/Synth.sol [hunk 5]
```diff
@@ -129,37 +107,15 @@ contract Synth is ExternStateToken {
         public
         optionalProxy
         returns (bool)
-    {
-        require(from != 0xfeefeefeefeefeefeefeefeefeefeefeefeefeef, "The fee address is not allowed");
```
Comment (jacko125) [inline contracts/Synth.sol]: Good removing this require as per Anton's comments

File: publish/deployed/kovan/deployment.json [hunk 6]
```diff
@@ -29,11 +29,11 @@
...
 		"ExchangeRates": {
 			"name": "ExchangeRates",
-			"address": "0x1ec58775e6169e475cCfC10168e5c945c6e56a4b",
+			"address": "0xE15f59e3806787a1Ef621a3595511DF7999E97D2",
 			"source": "ExchangeRates",
-			"link": "https://kovan.etherscan.io/address/0x1ec58775e6169e475cCfC10168e5c945c6e56a4b",
-			"timestamp": "2019-12-13T04:58:00.000Z",
-			"txn": "https://kovan.etherscan.io/tx/0x0a63d2c1ba51e62fe7fed403f55f693477a175c13be83e88330816fcc2aaa18e",
+			"link": "https://kovan.etherscan.io/address/0xE15f59e3806787a1Ef621a3595511DF7999E97D2",
+			"timestamp": "2019-12-10T04:51:08.000Z",
```
Comment (jacko125) [inline publish/deployed/kovan/deployment.json]: Need to get the new ExchangeRates for Kovan

**Pull request merged**